action/protocol/staking/handlers.go

```diff
@@ -892,6 +906,27 @@ func (p *Protocol) fetchBucketAndValidate(
 	return bucket, nil
 }
 
+func (p *Protocol) generateCandidateID(owner address.Address, height uint64, csm CandidateStateManager) (address.Address, error) {
+	isValidID := func(id address.Address) bool {
+		return csm.GetByIdentifier(id) == nil && csm.GetByOwner(id) == nil
+	}
+	if isValidID(owner) {
+		return owner, nil
+	}
+	h := append(owner.Bytes(), byteutil.Uint64ToBytesBigEndian(height)...)
+	for i := 0; i < 1000; i++ {
```

Comment:
can you quickly test how much time this 1000 loop will take? I'm thinking 128 should be enough to generate a diff ID

Reply:
```
goos: darwin
goarch: arm64
pkg: github.com/iotexproject/iotex-core/action/protocol/staking
BenchmarkLoop-10            2366            489271 ns/op           33397 B/op       1024 allocs/op
```
around 0.5ms, which exclude query from csm. I think it's acceptable